lib/spack/spack/build_systems/cmake.py

```diff
@@ -73,14 +68,12 @@ class CMakePackage(PackageBase):
 
     build_time_test_callbacks = ['check']
 
-    depends_on('cmake', type='build')
-
-    def build_type(self):
-        """Returns the correct value for the ``CMAKE_BUILD_TYPE`` variable
+    # https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
+    variant('build_type', default='RelWithDebInfo',
+            description='The build type to build',
+            values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
```

Comment:
ah well, these `debug`, `release`, `rel_with_deb_info`, `min_size_rel` do not look pretty... Probably keep it as it is now...